README.rst

```diff
@@ -0,0 +1,415 @@
+    gradients.
+
+5. Add ``hvd.BroadcastGlobalVariablesHook(0)`` to broadcast initial variable states from rank 0 to all other processes.
+    This is necessary to ensure consistent initialization of all workers when training is started with random weights or
+    restored from a checkpoint. Alternatively, if you're not using ``MonitoredTrainingSession``, you can simply execute
+    the ``hvd.broadcast_global_variables`` op after global variables have been initialized.
+
+6. Modify your code to save checkpoints only on worker 0 to prevent other workers from corrupting them.
+    This can be accomplished by passing ``checkpoint_dir=None`` to ``tf.train.MonitoredTrainingSession`` if
+    ``hvd.rank() != 0``.
+
+Example (see the `examples <examples/>`_ directory for full training examples):
+
+.. code-block:: bash
+
+    import tensorflow as tf
+    import horovod.tensorflow as hvd
+
+
+    # Initialize Horovod
+    hvd.init()
+
+    # Pin GPU to be used to process local rank (one GPU per process)
+    config = tf.ConfigProto()
+    config.gpu_options.visible_device_list = str(hvd.local_rank())
+
+    # Build model...
+    loss = ...
+    opt = tf.train.AdagradOptimizer(0.01 * hvd.size())
+
+    # Add Horovod Distributed Optimizer
+    opt = hvd.DistributedOptimizer(opt)
+
+    # Add hook to broadcast variables from rank 0 to all other processes during
+    # initialization.
+    hooks = [hvd.BroadcastGlobalVariablesHook(0)]
+
+    # Make training operation
+    train_op = opt.minimize(loss)
+
+    # Save checkpoints only on worker 0 to prevent other workers from corrupting them.
+    checkpoint_dir = '/tmp/train_logs' if hvd.rank() == 0 else None
+
+    # The MonitoredTrainingSession takes care of session initialization,
+    # restoring from a checkpoint, saving to a checkpoint, and closing when done
+    # or an error occurs.
+    with tf.train.MonitoredTrainingSession(checkpoint_dir=checkpoint_dir,
+                                           config=config,
+                                           hooks=hooks) as mon_sess:
+      while not mon_sess.should_stop():
+        # Perform synchronous training.
+        mon_sess.run(train_op)
+
+
+Running Horovod
+---------------
+
+The example commands below show how to run distributed training. See the `Running Horovod <docs/running.md>`_
+page for more instructions, including RoCE/InfiniBand tweaks and tips for dealing with hangs.
+
+1. To run on a machine with 4 GPUs:
+
+.. code-block:: bash
+
+    horovodrun -np 4 -H localhost:4 python train.py
+
+
+2. To run on 4 machines with 4 GPUs each:
+
+.. code-block:: bash
+
+    horovodrun -np 16 -H server1:4,server2:4,server3:4,server4:4 python train.py
+
+
+3. To run in Docker, see the `Horovod in Docker <docs/docker.md>`_ page.
+
+4. To run in Kubernetes, see `Kubeflow <https://github.com/kubeflow/kubeflow/tree/master/kubeflow/mpi-job>`_,
```

Comment:
For some reason (4) has different formatting than other sections.

Reply:
Solved, thanks!